Adapters/Vungle/ISVungleAdapter/ISVungleAdapter.m

```diff
@@ -102,74 +79,39 @@ - (NSString *)sdkName {
-        _interstitialPlacementIdToServerData            = [ConcurrentMutableDictionary dictionary];
-        _interstitialServerDataToSmashDelegate          = [ConcurrentMutableDictionary dictionary];
-        
-        // Banner
-        _bannerPlacementIdToSmashDelegate               = [ConcurrentMutableDictionary dictionary];
-        _bannerPlacementIdToSize                        = [ConcurrentMutableDictionary dictionary];
-        _bannerPlacementIdToViewController              = [ConcurrentMutableDictionary dictionary];
-        _bannerPlacementIdToBannerState                 = [ConcurrentMutableDictionary dictionary];
-        _bannerPlacementIdToServerData                  = [ConcurrentMutableDictionary dictionary];
-        _bannerServerDataToSmashDelegate                = [ConcurrentMutableDictionary dictionary];
     }
-    
+
     return self;
 }
 
 - (void)initSDKWithAppId:(NSString *)appId {
-    
     // add self to the init delegates only in case the initialization has not finished yet
     if ((_initState == INIT_STATE_NONE) || (_initState == INIT_STATE_IN_PROGRESS)) {
         [initCallbackDelegates addObject:self];
     }
-    
+
     static dispatch_once_t onceToken;
     dispatch_once(&onceToken, ^{
-        
         _initState = INIT_STATE_IN_PROGRESS;
 
-        #pragma clang diagnostic push
-        #pragma clang diagnostic ignored "-Wundeclared-selector"
-        [[VungleSDK sharedSDK] performSelector:@selector(setPluginName:version:)
-                                    withObject:@"ironsource"
-                                    withObject:[self version]];
-        #pragma clang diagnostic pop
-                
+        [VungleAds setIntegrationName:@"ironsource" version:[self version]];
         LogAdapterApi_Internal(@"appId = %@ adaptersDebug = %d", appId, [ISConfigurations getConfigurations].adaptersDebug);
-        
-        // set Vungle delegates
-        [[VungleSDK sharedSDK] setSdkHBDelegate:[ISVungleAdapterSingleton sharedInstance]];
-        [[VungleSDK sharedSDK] setDelegate:[ISVungleAdapterSingleton sharedInstance]];
-        
-        // set debug log
-        [[VungleSDK sharedSDK] setLoggingEnabled:[ISConfigurations getConfigurations].adaptersDebug];
-        
-        // initiate singelton
-        [[ISVungleAdapterSingleton sharedInstance] addFirstInitiatorDelegate:self];
-        
-        NSError *error;
-        
+
         // init Vungle sdk
-        if (![[VungleSDK sharedSDK] startWithAppId:appId
-                                             error:&error]) {
-            _initState = INIT_STATE_FAILED;
-            LogAdapterApi_Internal(@"Vungle SDK init failed - error = %@", error);
-        }
+        [VungleAds initWithAppId:appId completion:^(NSError * _Nullable error) {
+            if (error) {
+                LogAdapterApi_Internal(@"Vungle SDK init failed - error = %@", error);
+                [self initFailedWithError:error];
```

Comment:
Based on https://github.com/Vungle/levelplay-ios-adapters/pull/1/files#r991695009
if u delete the set u can directly call onNetworkInitCallbackFailed/onNetworkInitCallbackSuccess here in the completion block

Reply:
u also need to set the proper state.

Reply:
@YueVungle Same withe above comment. :)